docs/src/tutorials/tutorial_4.md

```diff
@@ -0,0 +1,329 @@
+# **Tutorial 4:** Do You Know Our Mascot? - Learn About Transitions And Subactions!
+
+You have learned a couple of cool features of Javis already. Now you're ready to finally meet our little mascot. Well actually you can't just see him, we have to create him first. 😄
+
+## Our goal
+
+Let's create a list of what we want first
+- a circular head
+- some hair
+- eyes
+- a nose
+- a moving mouth
+- he should say something
+
+## Learning Outcomes
+
+This tutorial should show you the power of subactions which can be used to let your objects appear and disappear as well as move on a finer grained scale.
+
+## Starting with the basics of SubAction
+
+The `ground` function should be familiar to you as well as the general structure of the `javis` function if you have seen the first [tutorial](tutorial_1.md).
+
+```julia
+using Javis
+
+function ground(args...)
+    background("white")
+    sethue("black")
+end
+
+function title(args...)
+    fontsize(20)
+    text("Our Mascot", Point(0, -200),
+        valign=:middle, halign=:center)
+end
+
+function face()
+    video = Video(500, 500)
+    javis(video, [
+        BackgroundAction(1:150, ground),
+        Action(title; subactions=[
+            SubAction(1:5, appear(:fade)),
+        ]),
+    ], pathname="jarvis.gif", framerate=15)
+end
+```
+
+A new part is the second `Action` which takes in a function and `subactions` as a keyword argument this time.
+
+If you're wondering where the `frames` are specified:
+- You can leave the frames blank as here and then the frames from the previous action are used. 
```

Comment:
I would suggest this phrasing:

```mkd
A new component introduced in this tutorial is shown in the second `Action`.
It is the kwarg `subactions` followed by the object `SubAction`.
A `SubAction` is invoked when you want to apply an additional action or provide an additional functionality to the original `Action`.
Although one could use standard an `Action` to achieve the same functionality via an `id`, `subactions` provide a much easier and more organized method that also improves readability.

With the invocation of `SubAction` in the `face` function, the function can be broken down as follows:
- We define our set the background for the animation using a `BackgroundAction`.
- Using the `title` function, we invoke an `Action` and then provide a `SubAction.`
  - The `SubAction` utilizes the `Javis.jl` provided function, `appear` to cause the title to slowly appear over the `UnitRange` of `1:5`. 

> **NOTE:** For a `SubAction` you can leave the frames arg blank. 
> The frames from the previous action are used. 
```

Reply:
Thanks I've changed parts of it as some were not correct but added your second paragraph as a summary.

The **note** should be `Action` not `SubAction`. We currently don't provide leaving the frames out for subactions.